### ISSUE TITLE
OCs cannot have master variants

### ISSUE THREAD

**Issue by sauloperez** [openfoodfoundation/openfoodnetwork#4251]
### What we should change and why (this is tech debt)

The [ProductsRenderer](https://github.com/openfoodfoundation/openfoodnetwork/blob/master/lib/open_food_network/products_renderer.rb), the class in charge of loading the products to show in a shopfront, uses [OrderCycleDistributedProducts](https://github.com/openfoodfoundation/openfoodnetwork/blob/908e1dfcafb7a4e7ce79f0f12a27305a62dbbae7/app/services/order_cycle_distributed_products.rb#L7) to filter out products without a variant, which is something that cannot happen since https://github.com/openfoodfoundation/openfoodnetwork/commit/63f3ede766216c27c460d02f2a440d5fbacb4371, released back in 2015.

**Therefore, `OrderCycleDistributedProducts` should be changed to skip that filtering and return only the distributed variants.**

This will also **require a data migration** to adapt old data as stated in https://github.com/openfoodfoundation/openfoodnetwork/pull/4147/files#r313818809

### Context

This is the result of the discussion in https://github.com/openfoodfoundation/openfoodnetwork/pull/4147#discussion_r313413640

### Impact and timeline

This removes very slow and pointless logic which should have a direct impact on https://github.com/openfoodfoundation/openfoodnetwork/issues/3853.

This issue is particularly important since last week we spotted there are various issues regarding Memcached that cause many shops to miss cache resulting on insane response times :see_no_evil: 

For this to be valid, we should see a reduction in latency outliers which in turn should reduce the percentile 99, which is now very high.  Check it out in https://app.datadoghq.com/apm/resource/rails/rack.request/9cd76a4669a387a3?durationEnd=0&durationStart=0&end=1568026582676&env=production&index=apm-search&maxPercentile=99&paused=false&start=1567421782676

**Comment by sauloperez** [comment]
The trace https://app.datadoghq.com/apm/trace/4289205600593835211?routing_hint=1568104862999&env=production&sort=time&colorBy=service&graphType=flamegraph&shouldShowLegend=true perfectly shows how a considerable amount of work of the 2s the job took to process are spent doing the useless work described in this issue.

It'd be amazing if we could know the distributor and OC ids that trace belongs to so we could compare before and after the improvement.

**Comment by luisramos0** [comment]
@sauloperez with #4271 I am fixing the description of the issue but not what's mentioned in the  comment, the code in your comment is a totally different beast :-O
Shall we open a new issue for it?

Because it has scoping and stock management involved, it will be complicated to re-write. Maybe we don't need to touch it in this iteration and put it in the overall performance backlog but not in this effort. At least I'd leave it to after #4040 #4041 and #4042

**Comment by sauloperez** [comment]
Oh :hankey: I thought it was a consequence of this same issue :thinking_face: Good call then.
